scheme/sevsnp/evidence_handler.go

```diff
@@ -0,0 +1,530 @@
+// Copyright 2025 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+
+package sevsnp
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/google/go-sev-guest/abi"
+	"github.com/google/go-sev-guest/proto/sevsnp"
+	"github.com/google/go-sev-guest/verify"
+	"github.com/google/go-sev-guest/verify/trust"
+	sevsnpParser "github.com/jraman567/go-gen-ref/cmd/sevsnp"
+	"github.com/veraison/corim/comid"
+	"github.com/veraison/corim/corim"
+	"github.com/veraison/ear"
+	"github.com/veraison/ratsd/tokens"
+	"github.com/veraison/services/handler"
+	"github.com/veraison/services/log"
+	"github.com/veraison/services/proto"
+)
+
+var (
+	ErrNoARK           = errors.New("missing ARK certificate in evidence")
+	ErrNoASK           = errors.New("missing ASK certificate in evidence")
+	ErrNoVEK           = errors.New("evidence must supply VLEK or VCEK")
+	ErrNoVCEK          = errors.New("VCEK is missing")
+	ErrNoVLEK          = errors.New("VLEK is missing")
+	ErrTAMismatch      = errors.New("evidence Trust Anchor (ARK) doesn't match the provisioned one")
+	ErrNoProvisionedTA = errors.New("missing provisioned Trust Anchor")
+	ErrNoProvisionedRV = errors.New("unrecognized evidence")
+	ErrBadSigningKey   = errors.New("bad signing key in attestation report")
+)
+
+const (
+	ReportSigningKeyVcek = 0
+	ReportSigningKeyVlek = 1
+)
+
+// EvidenceHandler implements the IEvidenceHandler interface for SEVSNP
+type EvidenceHandler struct {
+}
+
+// GetName returns the name of this evidence handler instance
+func (o EvidenceHandler) GetName() string {
+	return "sevsnp-evidence-handler"
+}
+
+// GetAttestationScheme returns the attestation scheme
+func (o EvidenceHandler) GetAttestationScheme() string {
+	return SchemeName
+}
+
+// GetSupportedMediaTypes returns the supported media types for the SEVSNP scheme
+func (o EvidenceHandler) GetSupportedMediaTypes() []string {
+	return EvidenceMediaTypes
+}
+
+func transformEvidenceToCorim(token *proto.AttestationToken) (*corim.UnsignedCorim, error) {
+	tsm, err := parseAttestationToken(token)
+	if err != nil {
+		return nil, err
+	}
+
+	reportProto, err := abi.ReportToProto(tsm.OutBlob)
+	if err != nil {
+		return nil, err
+	}
+
+	refValComid, err := sevsnpParser.ReportToComid(reportProto, 0)
+	if err != nil {
+		return nil, err
+	}
+
+	err = refValComid.Valid()
+	if err != nil {
+		return nil, err
+	}
+
+	refValCorim := corim.UnsignedCorim{}
```

Comment:
just check, if we need to initialise other mandatory part of CoRIM here as well...

Reply:
I'm setting the Profile and CoMID fields below. Did I miss any other mandatory fields?

Should I use the NewUnsignedCorim() method here instead?

Reply:
Just rename refValCorim to evCoRIM